plugins/modules/files/archive.py

```diff
@@ -453,6 +453,13 @@ def main():
                 except OSError as e:
                     errors.append(to_native(b_path))
 
+            for b_path in b_expanded_paths:
+                try:
+                    if os.path.isdir(b_path):
+                        shutil.rmtree(b_path)
```

Comment:
This also removes other files which were not added to the archive (`exclude_path` option).

How about the following:
1. Sort b_expanded_paths reversely, and iterate over it that way. That should make sure that child directories come before parent directories.
2. Also use `os.scandir(b_path)` to check whether the directory is empty (resp. contains more than `.` and `..`). If it does, skip the directory.
3. Use `os.rmdir`, which only removes the directory if they are empty.

Reply:
This led me to another thing which I'm not sure if this is a bug or not.
If you specify the folder in path, you cannot exclude a file in the folder.
```
# does not exclude
path: tmpdir
exclude_path: tmpdir/a.txt
```
```
# excludes
path: tmpdir/*
exclude_path: tmpdir/a.txt
```
So if this behaviour is expected and ok, then the current way of removing path when `remove: yes` is ok.

Reply:
This has been addressed on https://github.com/ansible-collections/community.general/issues/593 as not a bug.
What do you suggest to do now?

Reply:
Hmm. I guess in that case it is ok.